### ISSUE TITLE
Cannot able to Interpret Output of npz model File

### ISSUE THREAD

**Issue by shashwatwork** [mp2893/med2vec#16]
Hello Ed,
While testing Med2vec to MIMIC database cannot able to Interpret the output of the model file.Whether these are model weights or predicted neighbour visit.
![capture](https://user-images.githubusercontent.com/22785727/51821658-f42b6400-22ff-11e9-8d1b-16f22b4c0e74.PNG)
Please Try to clarify my doubt!
Thank You

**Comment by mp2893** [comment]
Hi Shashwat,

W_output and b_output are used in Med2Vec training phase only.
They are used to predict the neighboring visits.
The weights (embeddings) you might be interested  are W_emb since they are used to encode medical codes such as diagnosis codes or medication codes.

Best,
Ed

**Comment by shashwatwork** [comment]
Thanks, Ed! 
But Just to get a more clear picture-

W_emb - These are embedding used for encoding medical codes,So these encoding is based on Med2Vec Architecture.
W_outputs - These are predicted neighbouring visits

**Please Correct me if I am getting it wrong
What about W_hidden and b_hidden?**

Best ,
Shashwat

**Comment by mp2893** [comment]
W_output is not the predicted neighboring visits.
It is "used" to predict neighboring visits. It is W_s of Eq.2 in the paper, which is used right before the softmax layer.
W_hidden is used to derive the visit embeddings. It is W_v in Fig.2 of the paper. 
(A visit is a set of medical codes)
And, of course, W_emb is W_c in Fig.2 of the paper.
Hope this clarifies.

Best,
Ed

**Comment by shashwatwork** [comment]
Ok Ed! I got the point
Sorry for raising so many issues 
So how can I get the final output of Med2Vec.
**Actually I want to use the learned visit level and code level representation from Med2Vec.**
**My problem statement is to use these learned visit as well as code representation in RNN for predicting Type II diabetes**
Here are some questions-
1.So Is there any way to use these learned visit level and code level representations and how can I derive it without using any mathematical formula?
2. Is there any way to use these learned visit level and code level representations to build the model?

Thank You!

**Comment by mp2893** [comment]
1. The current implementation does not let you generate visit embeddings and code embeddings after you've trained the parameters (and I don't plan to update the code, since Theano is no longer maintained) So you will need to follow the equations in the paper to generate the embeddings yourself.
They are pretty straightforward actually. For visit embedding, all you need to do is format your data as described in README, then just follow lines 65, 66, 67. For code embedding, the i-th row of W_emb is the code embedding for the i-th code.

2. That's up to you. Whether you want to use the visit embeddings as input to the RNN, or you want to sum the code embeddings yourself and feed it to the RNN, you can try many different approaches.

**Comment by shashwatwork** [comment]
Thanks Ed!
